core/modules/entity/entity.theme.inc

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * @file
+ * Holds the Entity module's theme functions.
+ */
+
+/**
+ * Process variables for entity.tpl.php.
+ */
+function template_preprocess_entity(&$variables) {
+  $variables['view_mode'] = $variables['elements']['#view_mode'];
+  $entity_type = $variables['elements']['#entity_type'];
+  $variables['entity_type'] = $entity_type;
+  $entity = $variables['elements']['#entity'];
+  $variables[$entity_type] = $entity;
+  $info = entity_get_info($entity_type);
+
+  $variables['title'] = check_plain(entity_label($entity_type, $entity));
+
+  $uri = entity_uri($entity_type, $entity);
+  $variables['url'] = $uri && !empty($uri['path']) ? url($uri['path'], $uri['options']) : FALSE;
+
+  if (isset($variables['elements']['#page'])) {
+    // If set by the caller, respect the page property.
+    $variables['page'] = $variables['elements']['#page'];
+  }
+  else {
+    // Else, try to automatically detect it.
+    $variables['page'] = $uri && !empty($uri['path']) && $uri['path'] == $_GET['q'];
+  }
+
+  // Helpful $content variable for templates.
+  $variables['content'] = array();
+  foreach (element_children($variables['elements']) as $key) {
+    $variables['content'][$key] = $variables['elements'][$key];
+  }
+
+  if (!empty($info['fieldable'])) {
+    // Make the field variables available with the appropriate language.
+    field_attach_preprocess($entity_type, $entity, $variables['content'], $variables);
+  }
+  list(, , $bundle) = entity_extract_ids($entity_type, $entity);
+
+  // Gather css classes.
+  $variables['classes_array'][] = backdrop_html_class('entity-' . $entity_type);
```

Comment:
These two lines create $variables[`classes_array`]. However, `entity.tpl.php` uses $classes rather than $classes_array for the css class markup.

$variable['classes'] is created in `template_preprocess()` and is already available in this preprocess function. I suggest adding these css classes to `$variable['classes']` and getting rid of `$variable['classes_array']` in these two lines.

This will assure that `entity.tpl.php` will also receive a css class with [entityType]-[bundle] (which currently it doesn't since it never uses $classes_array).